### ISSUE TITLE
The troubles caused by multiple configuration dependencies

### ISSUE THREAD

**Issue by MEILIDEKCL** [Alluxio/alluxio#17509]
**Is your feature request related to a problem? Please describe.**
 `alluxio.master.journal.folder` default value is `${alluxio.work.dir}/journal`, if we don't set the first config, it will use the default value which related to other config.
And in alluxio-2.1, `alluxio.work.dir` default value is `${alluxio.home}` and `alluxio.home` can't be set by user.
So if we set `alluxio.work.dir` as `/`, and  `alluxio.master.journal.folder` will default to `//journal/`. 
In alluxio, it will use URI to parse the path, and in this case, the `alluxio.master.journal.folder` will be parsed to `/`. The specific reason is that when parsing a URI, it will determine the content from the first `//` to the new `/` previous as the host name, and then the path information will follow. So `//journal/` is parsed as `/`.
```java
  public static URI getJournalLocation() {
    String journalDirectory = ServerConfiguration.get(PropertyKey.MASTER_JOURNAL_FOLDER);
    if (!journalDirectory.endsWith(AlluxioURI.SEPARATOR)) {
      journalDirectory += AlluxioURI.SEPARATOR;
    }
    try {
      return new URI(journalDirectory);
    } catch (URISyntaxException e) {
      throw new RuntimeException(e);
    }
  }
```
And in the 2.1 version, it will cause serious problem because of the follow codes.
```java
@Override
  public void format() throws IOException {
    File journalPath = mConf.getPath();
    if (journalPath.isDirectory()) {
      org.apache.commons.io.FileUtils.cleanDirectory(mConf.getPath());
    } else {
      if (journalPath.exists()) {
        FileUtils.delete(journalPath.getAbsolutePath());
      }
      journalPath.mkdirs();
    }
  }
```

**Describe the solution you'd like**
Actually, in the later version, it will check the permission of the path.
```java
 @Override
  public void format() throws IOException {
    if (mPath.isDirectory()) {
      if (alluxio.util.io.FileUtils.isStorageDirAccessible(mPath.getPath())) {
        FileUtils.cleanDirectory(mPath);
      } else {
        throw new AccessDeniedException(mPath.getPath());
      }
    } else {
      if (mPath.exists()) {
        FileUtils.forceDelete(mPath);
      }
      if (!mPath.mkdirs()) {
        throw new AccessDeniedException(mPath.getPath());
      }
    }
  }
```
But, it is still inconsistent. We expected it as a folder in working dir like `//journal/`, but it actually is `/`.
To sum up, this situation will not cause serious problems because of the permission check. So, this approach is also feasible now, and the problem is not significant in the later version.

**Comment by jja725** [comment]
Hi thanks for raising the issue. If it's resolved in a newer version I would mark this as resolved and please try out a later version. And we would consider this problem when we redesign the property system. @LuQQiu FYI

**Comment by LuQQiu** [comment]
FYI @jenoudet

**Comment by jenoudet** [comment]
Thanks for bringing this to our attention. Given that this only affects a version that is quite old and only in a very narrow case, I believe it's safe to mark this issue resolved. Like @jja725 said, we will pay close attention to path resolution in property values when reworking the property system.

**Comment by MEILIDEKCL** [comment]
Yeah, that sounds good.
